### PR TITLE
[TEST]: Improvement: Switch rules/meters comparison

### PR DESCRIPTION
**Pull request by yuliiamir**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy [hunk 1]
```diff
@@ -317,22 +313,15 @@ class Server42IslRttSpec extends HealthCheckSpecification {
...
             }*.match.inPort.unique() == [newS42Port.toString()]
             assert rules.find {
                 it.cookie == Cookie.SERVER_42_ISL_RTT_OUTPUT_COOKIE
             }.instructions.applyActions.flowOutput == newS42Port.toString()
-            assert expect(rules.find { it.cookie == Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE }, sameBeanAs(s42IslRttTurningRule)
-                    .ignoring("byteCount")
-                    .ignoring("packetCount")
-                    .ignoring("durationSeconds")
-                    .ignoring("durationNanoSeconds"))
+            assert rules.find { it.cookie == Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE } == s42IslRttTurningRule
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy:324]: The same strategy was used for comparison FlowRuleEntity (EqualsAndHashCode(excludes = 'durationSeconds, durationNanoSeconds, packetCount, byteCount')
 Could not use FlowEntry as  the following fields are excluded:
  "durationSeconds", "durationNanoSeconds", "packetCount", "byteCount", "idleTimeout", "hardTimeout", "version"

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/FlowRuleEntity.groovy [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import org.openkilda.messaging.info.rule.FlowSwapFieldAction
+
+import groovy.transform.Canonical
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+
+@ToString(includeNames = true)
+@Canonical
+@EqualsAndHashCode(excludes = 'durationSeconds, durationNanoSeconds, packetCount, byteCount')
+class FlowRuleEntity {
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/FlowRuleEntity.groovy:18]: Added a new testing entity FlowRuleEntity that allows us to compare rules on the switch as usage of the FlowEntry excludes some required fields from verification("idleTimeout", "hardTimeout", "version"). 
```
@EqualsAndHashCode(exclude = {"durationSeconds", "durationNanoSeconds", "packetCount", "byteCount",
        "idleTimeout", "hardTimeout", "version"})
FlowEntry
```

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy [hunk 3]
```diff
@@ -123,9 +122,13 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
             assert mirrorPointsDetails[0].status == FlowPathStatus.ACTIVE.toString().toLowerCase()
             assert mirrorPointsDetails[0].mirrorPointId == mirrorPointPayload.mirrorPointId
         }
-        with(flow.retrieveMirrorPoints()) {
-            points.size() == 1
-            expect points[0], sameBeanAs(mirrorPointPayload)
```
Comment (yuliiamir) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy:128]: Also small updates for checking mirror details without using sameBeanAs.

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/SwitchRules.groovy [hunk 4]
```diff
@@ -72,18 +68,21 @@ class SwitchRules {
...
 
-    List<FlowEntry> getRules() {
-        northboundService.getSwitchRules(switchId).flowEntries
+    List<FlowRuleEntity> getServer42ISLRules() {
+        getRules().findAll { (new Cookie(it.cookie).getType() in [CookieType.SERVER_42_ISL_RTT_INPUT] ||
+                it.cookie in [Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE, Cookie.SERVER_42_ISL_RTT_OUTPUT_COOKIE]) }
+    }
+
+    List<FlowRuleEntity> getRules() {
+        northboundService.getSwitchRules(switchId).flowEntries.collect { new FlowRuleEntity(it) }
```
Comment (IvanChupin) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/SwitchRules.groovy:86]: From developer point of view this is quite unusual approach to have service level logic(northbound service, database) in the model classes.
I noticed that this is not the only one place in our func-test code with the similar approach, and we have similar logic in other model classes in func test packages.

Do not treat it as a call to action:)

**Review by IvanChupin — APPROVED** [review]
Great job!👍🏼

**Review by pablomuri — APPROVED** [review]

**Pull request merged**